mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala

```diff
@@ -175,6 +177,33 @@ class VectorsSuite extends FunSuite {
     assert(v.size === x.rows)
   }
 
+  test("sqdist") {
+    val random = new Random(System.nanoTime())
+    for (m <- 1 until 1000 by 10) {
+      val nnz = random.nextInt(m) + 1
+
+      val indices1 = random.shuffle(0 to m - 1).toArray.slice(0, nnz).sorted
```

Comment:
@mengxr @jkbradley Please refer to [https://github.com/scala/scala/commit/51ec62a8c307cd66fe34fe07c85100d48b54aa8f](https://github.com/scala/scala/commit/51ec62a8c307cd66fe34fe07c85100d48b54aa8f). It is caused by the problem in scala's type constructor inference and the creation of some Abstract* classes in scala 2.10.